### PR TITLE
feat: series and expr quantile

### LINKED ISSUE #203: Add Expr.quantile and Series.quantile
https://docs.pola.rs/py-polars/html/reference/expressions/api/polars.Expr.quantile.html#polars.Expr.quantile

I'm _slightly_ worried that the default interpolation method might change in Polars at some point, so we should probably not put any default for it in Narwhals and force users to specify it explicitly

### PR DESCRIPTION
**Pull request by FBruzzesi**
# Description

Fix #203 by introducing `.quantile(...)` for Expr and Series.

I am opening this as draft because of two main reasons:

- [ ] I may need some support with type hinting
- [ ] Need decision for how to deal with non numeric formats

### CODE REVIEW

File: docs/api-reference/expressions.md [hunk 1]
```diff
@@ -26,11 +26,12 @@
...
         - n_unique
         - over
-        - unique
+        - quantile
         - sample
         - shift
         - sort
         - std
         - sum
+        - unique
```
Comment (FBruzzesi) [inline docs/api-reference/expressions.md:35]: I assumed it was supposed to be alphabetically sorted

**Comment by FBruzzesi** [comment]
To check the pandas behavior in the [failing GHA](https://github.com/narwhals-dev/narwhals/actions/runs/9224134064/job/25378803025?pr=207), I found that the min version in which it doesn't break is `pandas==1.3.5`. Pandas versions previous to that one require `numpy<=1.21`.

**Comment by MarcoGorelli** [comment]
thanks - maybe we can just pin `numpy` in the ci-minimum-versions CI job too then?

File: narwhals/expression.py [hunk 2]
```diff
@@ -1417,6 +1418,54 @@ def is_last_distinct(self) -> Expr:
...
+
+            >>> func(df_pl)  # doctest: +NORMALIZE_WHITESPACE
+            shape: (1, 2)
+            ┌──────┬──────┐
+            │ a    ┆ b    │
+            │ ---  ┆ ---  │
+            │ f64  ┆ f64  │
+            ╞══════╪══════╡
+            │ 25.0 ┆ 75.0 │
+            └──────┴──────┘
```
Comment (MarcoGorelli) [inline narwhals/expression.py:1463]: i'm a little surprised that they output different results, have you checked why?

**Review by MarcoGorelli — COMMENTED** [review]
thanks @FBruzzesi ! just got a question

File: narwhals/expression.py (hunk 2 above)
Comment (FBruzzesi) [reply narwhals/expression.py:1463]: True! The quantile falls in the middle of two numbers. numpy and pandas implementation of `method="nearest"` seem to round down while rust/polars to round up? 
Should we use a more stable example?

Comment (MarcoGorelli) [reply narwhals/expression.py:1463]: yeah maybe 'linear' in the examples, and then add a note explaining that there may be some slight implementation differences when rounding?

**Pull request merged**